crates/redeem/src/lib.rs

```diff
@@ -480,7 +480,6 @@ impl<T: Config> Pallet<T> {
             .ok_or(Error::<T>::ArithmeticOverflow)?;
 
         let amount_wrapped_in_collateral = ext::oracle::wrapped_to_collateral::<T>(vault_to_be_burned_tokens)?;
-        let punishment_fee_in_collateral = ext::fee::get_punishment_fee::<T>(amount_wrapped_in_collateral)?;
```

Comment:
Don't we still want this?

Reply:
Please double check this, but the behavior of cancel_redeem should remain unchanged by this pr. The only difference is that previously we had split it up in 2 parts, because the user would get 10% and the fee pool the rest. But now both parts are transferred to the vault, so the logic can be simplified

Reply:
I think this does change it since the punishment fee is on-top of the slashed amount, see: https://github.com/interlay/interbtc/blob/master/crates/fee/src/lib.rs#L152-L154

Reply:
But `calculate_slashed_amount` [accounts](https://github.com/interlay/interbtc/blob/407a5c6e9bbcdc6cefafd16eec18776ca6070b6a/crates/sla/src/lib.rs#L287-L292) for that, no?

Reply:
From what I can see the `PunishmentFee` is unused so I'm not sure where we would calculate the 10%. Anyway the plan is to remove the SLA module so we can fix this in another PR.

Reply:
It's done in vault_registry::calculate_slashed_amount, which calls sla::calculate_slashed_amount